### PR TITLE
iox-#1196 Fix clang tidy warnings in `RelativePointerData`

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

* Fix `clang-tidy` warnings in `RelativePointerData`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1196 (partly)

### CODE REVIEW

**Review by elfenpiff — CHANGES_REQUESTED** [review]
It took my now 2 full days to fix all the overlooked warnings in cxx. So I made up a new rule, could you please add all files which you fixed to `/clang-tidy-files-to-scan.txt`

When the CI approves that everything is fixed I will approve as well.
Sorry for add this little burden on you but we have to be more efficient.

**Review by elBoberido — APPROVED** [review]

File: .clang-tidy-diff-scans.txt [hunk 1]
```diff
@@ -2,7 +2,10 @@
 ./iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/*
 ./iceoryx_hoofs/test/moduletests/test_cxx*
 ./iceoryx_hoofs/source/cxx/*
-
+./iceoryx_hoofs/source/relocatable_pointer/relative_pointer_data.cpp
+./iceoryx_hoofs/test/moduletests/test_relative_pointer_data.cpp
```
Comment (elfenpiff) [inline .clang-tidy-diff-scans.txt:6]: This file is not changed by this PR. @mossmaurice did you perform a manual clang-tidy-15 scan?

Comment (elBoberido) [reply .clang-tidy-diff-scans.txt:6]: @elfenpiff I guess the reason is that it is a textbook example on how to write tests ... hmm, where does this stinky smell suddenly come from ... must be self-laudation :laughing: ... sorry, couldn't resist

**Review by elBoberido — APPROVED** [review]

**Review by MatthiasKillat — COMMENTED** [review]
Looks good. There seems to be a conflict with `.clang-tidy-diff-scans.txt` though.

File: .clang-tidy-diff-scans.txt (hunk 1 above)
Comment (elfenpiff) [reply .clang-tidy-diff-scans.txt:10]: Before I approve: @mossmaurice did you perform a manual clang-tidy-15 scan?

**Review by elBoberido — APPROVED** [review]

**Review by MatthiasKillat — APPROVED** [review]

**Pull request merged**